tap_recharge/discover.py

```diff
@@ -1,20 +1,46 @@
-from singer.catalog import Catalog, CatalogEntry, Schema
-from tap_recharge.schema import get_schemas, STREAMS
+from singer.catalog import Catalog
+from tap_recharge.schema import get_schemas
+
+def _get_key_properties_from_meta(schema_meta: list) -> str:
+    """
+    Gets the table-key-properties from the schema metadata.
+    """
+    return schema_meta[0].get('metadata').get('table-key-properties')
```

Comment:
It's a little surprising this is always the stream metadata

Reply:
This should work since the tap is using `singer.metadata.get_standard_metadata` (it builds a dict and then converts to a list) to create the metadata and since Python 3.6 dict's are ordered so the list always has the first entry as the metadata.

These functions have been removed as part of the extraneous keys below, so it's moot.